File: app/src/main/java/com/adasa/foodplan/domain/model/MealPlan.kt
package com.adasa.foodplan.domain.model

import kotlinx.datetime.Clock
import kotlinx.datetime.LocalDate
import kotlinx.datetime.TimeZone
import kotlinx.datetime.todayIn

data class MealSlot(
    val type: MealCategory = MealCategory.LUNCH,
    val recipeId: String = ""
)

data class DailyGoal(
    val kcalTarget: Int = 1350,
    val proteinTarget: Int = 120
)

data class DayPlan(
    val id: String = "",
    val date: LocalDate = Clock.System.todayIn(TimeZone.currentSystemDefault()),
    val meals: List<MealSlot> = emptyList(),
    val proteinPowderGrams: Double = 0.0,
    val goal: DailyGoal = DailyGoal()
)

data class MealPlan(
    val id: String = "",
    val name: String = "",
    val startDate: LocalDate = Clock.System.todayIn(TimeZone.currentSystemDefault()),
    val endDate: LocalDate = Clock.System.todayIn(TimeZone.currentSystemDefault()),
    val days: List<DayPlan> = emptyList()
)

File: flutter-app/lib/domain/model/meal_plan.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'recipe.dart';

part 'meal_plan.freezed.dart';

// ── Domain models ─────────────────────────────────────────────────────────────

@freezed
class MealPlan with _$MealPlan {
  const factory MealPlan({
    required String id,
    required String name,
    required DateTime startDate,
    required DateTime endDate,
    @Default(<DayPlan>[]) List<DayPlan> days,
  }) = _MealPlan;
}

@freezed
class DayPlan with _$DayPlan {
  const factory DayPlan({
    required String id,
    required DateTime date,
    @Default(<MealSlot>[]) List<MealSlot> meals,
    @Default(0.0) double proteinPowderGrams,
    @Default(DailyGoal()) DailyGoal goal,
  }) = _DayPlan;
}

@freezed
class MealSlot with _$MealSlot {
  const factory MealSlot({
    required MealCategory type,
    required String recipeId,
  }) = _MealSlot;
}

@freezed
class DailyGoal with _$DailyGoal {
  const factory DailyGoal({
    @Default(1350.0) double kcalTarget,
    @Default(120.0) double proteinTarget,
  }) = _DailyGoal;
}
